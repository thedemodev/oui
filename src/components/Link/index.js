import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Link = ({
  ariaLabel,
  children,
  href,
  onClick,
  isDisabled = false,
  newWindow = false,
  style,
  testSection,
}) => {
  const classes = classNames({
    ['link']: true,
    [`link--${style}`]: style,
    ['link--disabled']: isDisabled,
    'pointer-events--none': isDisabled,
  });

  if (!href && !onClick) {
    return (
      <span
        data-oui-component={ true }
        className={ classes }
        data-test-section={ testSection }
        data-track-id={ testSection }
        disabled={ isDisabled }>
        { children }
      </span>
    );
  }

  return (
    <a
      aria-label={ ariaLabel }
      data-oui-component={ true }
      href={ href }
      className={ classes }
      data-test-section={ testSection }
      data-track-id={ testSection }
      onClick={ onClick }
      { ...(newWindow ? { target: '_blank', rel: 'noopener noreferrer' } : {}) }
      disabled={ isDisabled }>
      { children }
    </a>
  );
};

Link.propTypes = {
  /** Label to use for screen readers */
  ariaLabel: PropTypes.string,
  /** Text/node that is linked */
  children: PropTypes.node.isRequired,
  /** Href for the anchor element */
  href: PropTypes.string,
  /** Disable link and interactivity */
  isDisabled: PropTypes.bool,
  /** Open link in new window */
  newWindow: PropTypes.bool,
  /** Click handler function */
  onClick: PropTypes.func,
  /** Link style options */
  style: PropTypes.oneOf([
    'default',
    'dark',
    'muted',
    'bad-news',
    'reverse',
  ]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
};

export default Link;
