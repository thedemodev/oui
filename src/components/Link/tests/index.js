import React from 'react';
import Link from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/Link', () => {

  it('should render disabled', () => {
    const output = shallow(
      <Link
        href='http://google.com'
        isDisabled={ true }>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render target blank attribute', () => {
    const output = shallow(
      <Link
        href='http://google.com'
        newWindow={ true }>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render style classes', () => {
    const output = shallow(
      <Link
        href='http://google.com'
        style='muted'>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render as a span if there is no href', () => {
    const output = shallow(
      <Link>Faux Link</Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render ariaLabel', () => {
    const output = shallow(
      <Link
        ariaLabel="Go to Google"
        href='http://google.com'
        style='muted'>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
