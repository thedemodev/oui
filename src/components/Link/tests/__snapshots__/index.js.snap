// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Link should render ariaLabel 1`] = `
<a
  aria-label="Go to Google"
  className="link link--muted"
  data-oui-component={true}
  disabled={false}
  href="http://google.com"
>
  Test Link
</a>
`;

exports[`components/Link should render as a span if there is no href 1`] = `
<span
  className="link"
  data-oui-component={true}
  disabled={false}
>
  Faux Link
</span>
`;

exports[`components/Link should render disabled 1`] = `
<a
  className="link link--disabled pointer-events--none"
  data-oui-component={true}
  disabled={true}
  href="http://google.com"
>
  Test Link
</a>
`;

exports[`components/Link should render style classes 1`] = `
<a
  className="link link--muted"
  data-oui-component={true}
  disabled={false}
  href="http://google.com"
>
  Test Link
</a>
`;

exports[`components/Link should render target blank attribute 1`] = `
<a
  className="link"
  data-oui-component={true}
  disabled={false}
  href="http://google.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Test Link
</a>
`;
